src/aesophia_cli.erl

```diff
@@ -292,21 +289,23 @@ write_aci(OutFile, ACI) ->
     file:write_file(OutFile, ACI).
 
 %% Maybe better to do on the compiler side...
-prepare_args("") ->
-    {ok, []};
-prepare_args(ArgsStr) ->
-    try aeso_parser:string("contract C =\n  function foo() = (" ++ ArgsStr ++ ")") of
-        [{contract, _, _, [{letfun, _, _, _, _, Args}]}] ->
-            case Args of
-                {tuple, _, Args1} -> {ok, [prepare_arg(Arg) || Arg <- Args1]};
-                _                 -> {ok, [prepare_arg(Args)]}
-            end
+prepare_call(Call) ->
+    try aeso_parser:string("contract C =\n function foo() = " ++ Call) of
```

Comment:
But if Call is "foo()" or the example above "foo(42, true)" then the choice of the name foo is a bit of a handicap, I guess.

Reply:
ok. got it.

Reply:
We only parse it, it doesn't have to make sense :-)